File: security-platform/agents/dotnet/SecurityPlatform.Agent/RedactionRule.cs
namespace SecurityPlatform.Agent
{
    /// <summary>
    /// Redaction rule configuration.
    /// </summary>
    public class RedactionRule
    {
        public string? Pattern { get; set; }
        public string? Replacement { get; set; }
        public string? Field { get; set; }
    }
}

File: security-platform/agents/ios/Sources/SecurityPlatform/RedactionRule.swift
import Foundation

/// Redaction rule configuration.
public struct RedactionRule {
    public var pattern: String
    public var replacement: String?
    public var field: String?

    public init(pattern: String, replacement: String? = nil, field: String? = nil) {
        self.pattern = pattern
        self.replacement = replacement
        self.field = field
    }
}
